source/dubregistry/api.d

```diff
@@ -24,13 +24,32 @@ void registerDubRegistryAPI(URLRouter router, DubRegistry registry)
 {
 	auto pkgs = new LocalDubRegistryAPI(registry);
 	router.registerRestInterface(pkgs, "/api");
+	router.get("/api/packages/dump", delegate(req, res) => dumpPackages(req, res, registry));
 }
 
 /// Compatibility alias.
 deprecated("Use registerDubRegistryAPI instead.")
 alias registerDubRegistryWebApi = registerDubRegistryAPI;
 
 
+private void dumpPackages(HTTPServerRequest req, HTTPServerResponse res, DubRegistry registry)
+{
+	import vibe.data.json : serializeToPrettyJson;
+	import vibe.stream.wrapper : StreamOutputRange;
+
+	res.contentType = "application/json; charset=UTF-8";
+	res.headers["Content-Encoding"] = "gzip"; // force GZIP compressed response
+	auto dst = StreamOutputRange(res.bodyWriter);
+	dst.put('[');
+	bool first = true;
+	foreach (p; registry.getPackageDump()) {
+		if (!first) dst.put(',');
+		else first = false;
+		serializeToPrettyJson(&dst, p);
```

Comment:
I checked and the overhead in the compressed output is negligible (the formatting gets compressed perfectly because it is exactly the same for every entry). On the other hand this makes working with the dump a lot more practical (you'd have to find a JSON formatter or viewer that can cope with large files first).

Using BSON would mostly just complicate the client requirements and is probably not more compact in the end (the biggest advantage of BSON is being able to store compact binary data and that jumping over whole values is O(1)). But in a future revision, letting the client control this with an "Accept" header would be an option.